datafusion/sqllogictest/test_files/subquery.slt

```diff
@@ -1028,6 +1028,80 @@ false
 true
 true
 
+# in_subquery_to_join_with_correlated_outer_filter_disjunction
```

Comment:
Could you also add `EXPLAIN` tests for these queries too so we can review if the plans will generate the correct results?

Reply:
Added in https://github.com/apache/datafusion/pull/12945/commits/ea5d04e7ae14a3b9fb3554033161e8ce59fd1f6f